### ISSUE TITLE
Configuring metrics used for memory profiling

### ISSUE THREAD

**Issue by pellared** [signalfx/gdi-specification#203]
In SFx .NET (but later we will need also to support it in OTel distribution) we are asked to introduce a setting to control what "metric sets" should be enabled. Our current "metrics sets" would be: .NET runtime metrics, process metrics, ASP.NET Core metrics.

Is it possible to configure similar things in Java and Node.js? If so then how?

In [OTel .NET Auto](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumentations) there is a `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS`, but it is not stable at all.

CC @mateuszrzeszutek @seemk @signalfx/gdi-dotnet-approvers

**Comment by mateuszrzeszutek** [comment]
In the javaagent it works exactly the same as any other instrumentation: you can disable it using [this config option](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-agent-instrumentation), and our custom metrics instrumentation names are [here](https://github.com/signalfx/splunk-otel-java/blob/main/docs/metrics.md#supported-libraries).

**Comment by pellared** [comment]
@mateuszrzeszutek Thanks a lot 👍 AFAIK `OTEL_INSTRUMENTATION_[NAME]_ENABLED` is not in the OTel spec (at least I have not found it). I could't also find other languages have some similar env var... I think we could use the same pattern in OTel .NET for consistency. Is it something worth brining the OTel spec?

**Comment by mateuszrzeszutek** [comment]
Definitely! Though I feel that until now the spec has been kinda avoiding the topic of instrumentation configs; pretty much only SDK configuration options are defined there.

**Comment by pellared** [comment]
@mateuszrzeszutek Is there an option in OTel Java to enable the instrumentations in a "allow list" way. Meaning: disable all and explicitly enable the ones that the user needs?

**Comment by mateuszrzeszutek** [comment]
Yeah, it's kinda mentioned [here](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#enable-manual-instrumentation-only) -- you can disable every instrumentation by default `Dotel.instrumentation.common.default-enabled=false` and then just cherry-pick.

**Comment by pellared** [comment]
@mateuszrzeszutek  Via env var it would be `OTEL_INSTRUMENTATION_COMON_DEFAULT_ENABLED=false`?

**Comment by mateuszrzeszutek** [comment]
Yep

**Comment by pellared** [comment]
Closing. I will try to continue first via OTel .NET AutoInstrumentation SIG and then OTel Spec.

@mateuszrzeszutek thanks a lot
